templates/flamegraph.mustache

```diff
@@ -73,16 +75,22 @@ function init() {
         draw();
     }
 
-    window.excimerChart.label(function(d) {
-        return d.data.name + ' (' + ((d.x1 - d.x0) * 100).toFixed(3)
-                + '%, ' + d.value + ' samples, duration ~' +  d.value * {{samplerate}} + ' ms)'
-    });
+    displaysamplestats = function (d) {
+        time = d.value * {{samplerate}};
+        seconds = time / 1000;
+        minutes = Math.floor(seconds / 60);
+        hours = Math.floor(minutes / 60);
+        minutes = minutes % 60;
+        seconds = (seconds % 60).toFixed(3);
+        durationstr = (hours ? (hours + 'h ') : '') + (minutes ? (minutes + 'm ') : '') + seconds + 's';
 
-    var tip = flamegraph.tooltip.defaultFlamegraphTooltip().text(function(d) {
         return d.data.name + ' (' + ((d.x1 - d.x0) * 100).toFixed(3)
-        + '%, ' + d.value + ' samples, duration ~' +  d.value * {{samplerate}} + ' ms)'
-    });
+                + '%, ' + d.value.toLocaleString("{{lang}}") + ' samples, duration ~ ' +  durationstr + ')'
```

Comment:
Can we please clean this up to make it easier to read and make future changes?

```suggestion
        const percentage = ((d.x1 - d.x0) * 100).toFixed(3)
        const sampleCount = d.value.toLocaleString("{{lang}}")
        return d.data.name + ` (${percentage}%, ${sampleCount} samples, duration ~ ${durationstr})`
```

Reply:
done